casepro/statistics/migrations/0001_initial.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+from django.conf import settings
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('orgs', '0016_taskstate_is_disabled'),
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('cases', '0037_partner_is_restricted'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='DailyOrgCount',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('type', models.CharField(max_length=1)),
+                ('day', models.DateField(help_text='The day this count is for')),
+                ('count', models.PositiveIntegerField(default=1)),
+                ('squashed', models.BooleanField(default=False)),
+                ('org', models.ForeignKey(to='orgs.Org')),
+            ],
+            options={
+                'abstract': False,
+            },
+        ),
+        migrations.CreateModel(
+            name='DailyUserCount',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('type', models.CharField(max_length=1)),
+                ('day', models.DateField(help_text='The day this count is for')),
+                ('count', models.PositiveIntegerField(default=1)),
+                ('squashed', models.BooleanField(default=False)),
+                ('org', models.ForeignKey(to='orgs.Org')),
+                ('user', models.ForeignKey(to=settings.AUTH_USER_MODEL)),
+            ],
+            options={
+                'abstract': False,
+            },
+        ),
+        migrations.CreateModel(
+            name='DailyPartnerCount',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('type', models.CharField(max_length=1)),
+                ('day', models.DateField(help_text='The day this count is for')),
+                ('count', models.PositiveIntegerField(default=1)),
+                ('squashed', models.BooleanField(default=False)),
+                ('partner', models.ForeignKey(to='cases.Partner')),
+            ],
```

Comment:
I do like the idea of the slug field for determining which values should be squashed, but it seems like I'd still want foreign keys for org/partner/user for fast fetching of these counts. 

I also need to do label-based counts, so if I'm trying to do this with one table, I end up with with 4 foreign keys, 3 of which are null on each row, and then starts to look like something which should be normalised. Hmm..